libgadget/cooling_uvfluc.c

```diff
@@ -238,16 +238,19 @@ static struct UVBG get_local_UVBG_from_J21(double redshift, int heiiionized, dou
     return uvbg;
 }
 
-//placeholder function so i don't have to delete old get_local_UVBG yet
+//switch function that decides whether to use excursion set or global UV background
+/*TODO: there are a few continuity issues to consider fixing.
+ * switching to the global rates for heii ionised particles is discontinuous.
+ * if the z_reion tables provided finish after ExcursionSetZStop, particles could rapidly recombine.
+ * I'm not sure how initial heating from reionisation (see D'Aloisio et al. 2019) is handled here */
 struct UVBG get_local_UVBG(double redshift, double * Pos, const double * PosOffset, int heiiionized, double J21)
 {
-    if(All.ExcursionSetFlag && (redshift > All.ExcursionSetZStop))
+    if(All.ExcursionSetReionOn && (redshift > All.ExcursionSetZStop) && !(heiiionized))
```

Comment:
I think just assume (or enforce) that All.ExcursionSetZstop < the helium reionisation redshift and so you don't need to worry about it.

Reply:
Fair enough, I'll remove the heiiionized flag and use the highest of the two redshifts to stop the excursion set.

Reply:
Thanks - given your comment above it sounds like we don't really do that correctly anyway, so we may as well be clear about that